LPD-45806 Upgrade to Checkstyle 10.21.2

ling-alan-huang opened the pull request.

ling-alan-huang committed:
LPD-45806 Rename

see https://github.com/checkstyle/checkstyle/commit/24ead56c

ling-alan-huang committed:
LPD-45806 Remove it since it conflicts with JavaPaser and MissingParenthesesCheck

ling-alan-huang committed:
LPD-45806 Upgrade to Checkstyle 8.43

ling-alan-huang committed:
LPD-45806 Upgrade to antlr4-runtime 4.9.2 to avoid warning

Console: [java] ANTLR Tool version 4.9.2 used for code generation does not match the current runtime version 4.8ANTLR Runtime version 4.9.2 used for parser compilation does not match the current runtime version 4.8ANTLR Tool version 4.9.2 used for code generation does not match the current runtime version 4.8ANTLR Runtime version 4.9.2 used for parser compilation does not match the current runtime version 4.8

ling-alan-huang committed:
LPD-45806 Auto SF

ling-alan-huang committed:
LPD-45806 Generated

ling-alan-huang committed:
LPD-45806 Add more modifers to support Java 17

ling-alan-huang committed:
LPD-45806 Follow the order in JLS

ling-alan-huang committed:
LPD-45806 Support new modifiers when parsing Java class

ling-alan-huang committed:
LPD-45806 Add modifiers in JavaClass

ling-alan-huang committed:
LPD-45806 Parse sealed, non-sealed, and permits

ling-alan-huang committed:
LPD-45806 Rename, they are reserved words not key words

ling-alan-huang committed:
LPD-45806 Parse permits

ling-alan-huang committed:
LPD-45806 Do not use seaeld/non-seald

ling-alan-huang committed:
LPD-45806 Add test case

ling-alan-huang committed:
LPD-45806 Generated

ling-alan-huang committed:
LPD-45806 Upgrade to antlr4-runtime:4.13.2

ling-alan-huang committed:
LPD-45806 Make it runtimeOnly

ling-alan-huang committed:
LPD-45806 Upgrade to Checkstyle 10.21.2

ling-alan-huang committed:
LPD-45806 Remove antlr.* in portal-tools-java-parser

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

ling-alan-huang committed:
LPD-45806 Remove antlr.* source-formatter

ling-alan-huang committed:
LPD-45806 Fix for difference AST in Checkstyle-10.21.2

Example: catch (AMImageConfigurationException | IOException exception) {

Checkstyle-8.29
`--LITERAL_CATCH -> catch [22:2]
    |--LPAREN -> ( [22:8]
    |--PARAMETER_DEF -> PARAMETER_DEF [22:39]
    |   |--MODIFIERS -> MODIFIERS [22:39]
    |   |--TYPE -> TYPE [22:39]
    |   |   `--BOR -> | [22:39]
    |   |       |--IDENT -> AMImageConfigurationException [22:9]
    |   |       `--IDENT -> IOException [22:41]
    |   `--IDENT -> exception [22:53]

Checkstyle-10.21.2
`--LITERAL_CATCH -> catch [22:2]
    |--LPAREN -> ( [22:8]
    |--PARAMETER_DEF -> PARAMETER_DEF [22:9]
    |   |--MODIFIERS -> MODIFIERS [22:9]
    |   |--TYPE -> TYPE [22:9]
    |   |   |--IDENT -> AMImageConfigurationException [22:9]
    |   |   |--BOR -> | [22:39]
    |   |   `--IDENT -> IOException [22:41]

ling-alan-huang commented:
```
com.puppycrawl.tools.checkstyle.api.CheckstyleException: 处理时抛出异常 /Users/alan/liferay_code/master/liferay-portal/modules/apps/account/account-test/src/testIntegration/java/com/liferay/account/service/test/AccountEntryUserRelLocalServiceTest.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:312)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:226)
	at com.liferay.source.formatter.processor.BaseSourceProcessor.processCheckstyle(BaseSourceProcessor.java:467)
	at com.liferay.source.formatter.processor.JavaSourceProcessor._processCheckstyle(JavaSourceProcessor.java:229)
	at com.liferay.source.formatter.processor.JavaSourceProcessor.postFormat(JavaSourceProcessor.java:101)
	at com.liferay.source.formatter.processor.BaseSourceProcessor.format(BaseSourceProcessor.java:117)
	at com.liferay.source.formatter.SourceFormatter._runSourceProcessor(SourceFormatter.java:1306)
	at com.liferay.source.formatter.SourceFormatter$1.call(SourceFormatter.java:400)
	at com.liferay.source.formatter.SourceFormatter$1.call(SourceFormatter.java:396)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.NullPointerException: Cannot invoke "com.puppycrawl.tools.checkstyle.api.DetailAST.getType()" because "firstChildDetailAST" is null
	at com.liferay.source.formatter.checkstyle.check.MissingParenthesesCheck.doVisitToken(MissingParenthesesCheck.java:41)
	at com.liferay.source.formatter.checkstyle.check.BaseCheck.visitToken(BaseCheck.java:86)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:393)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:466)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:331)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:211)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:101)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:340)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:299)
	... 13 more
```
